docs/building-modules/keeper.md

```diff
@@ -0,0 +1,89 @@
+# Keepers
+
+## Pre-requisite Reading
+
+- [Introduction to SDK Modules](./intro.md)
+
+## Synopsis
+
+`Keeper`s refer to a Cosmos SDK abstraction whose role is to manage access to the subset of the state defined by various modules. `Keeper`s are module-specific, i.e. the subset of state defined by a module can only be accessed by a `keeper` defined in said module. If a module needs to access the subset of state defined by another module, a reference to the second module's internal `keeper` needs to be passed to the first one. 
+
+- [Motivation](#motivation)
+- [Type Definition](#type-definition)
+- [Implementing Methods](#implementing-methods)
+
+## Motivation
+
+The Cosmos SDK is a framework that makes it easy for developers to build complex decentralised applications from scratch, mainly by composing modules together. As the ecosystem of open source modules for the Cosmos SDK expands, it will become increasingly likely that some contain vulnerabilities, as a result of the negligence or malice of their developer. 
+
+The Cosmos SDK adopts a [capabilities-based approach](./ocap.md) to help developers better protect their application from unwanted inter-modules interactions, and `keeper`s are at the core of this approach. A `keeper` can be thougt quite literally as the gatekeeper of a module's store(s). Each store defined within a module comes with a `storeKey`, which grants unlimited access to it. The module's `keeper` holds this `storeKey` (which should otherwise remain unexposed), and defines [methods](#implementing-methods) for reading and writting to the store(s). 
+
+The core idea behind the object-capabilities approach is to only reveal what is necessary to get the work done. In practice, this means that instead of handling permissions of modules through an access-control lists, the `keeper`s of modules get a reference to a specific instance of the other modules' `keeper`s they need access to. As a consequence, a module can only interract with the subset of state defined in another module via the methods exposed by the instance of the other module's `keeper`. This is a great way for developers to control the interactions that their own module have with modules developed by external developers. 
```

Comment:
```suggestion
The core idea behind the object-capabilities approach is to only reveal what is necessary to get the work done. In practice, this means that instead of handling permissions of modules through access-control lists, module `keeper`s are passed a reference to the specific instance of the other modules' `keeper`s that they need to access. As a consequence, a module can only interact with the subset of state defined in another module via the methods exposed by the instance of the other module's `keeper`. This is a great way for developers to control the interactions that their own module can have with modules developed by external developers. 
```